src/components/notifications/AppSelector/index.tsx

```diff
@@ -16,52 +16,66 @@ import Text from '../../general/Text'
 import MobileHeader from '../../layout/MobileHeader'
 import { AnimatePresence, motion } from 'framer-motion'
 import { handleImageFallback } from '../../../utils/ui'
-import { getDbEchoRegistrations } from '../../../utils/idb'
+
+const SUBSCRIPTION_LOADER_TIMEOUT = 3000
 
 const AppSelector: React.FC = () => {
   const { pathname } = useLocation()
   const [search, setSearch] = useState('')
   const isMobile = useIsMobile()
-  const [dropdownToShow, setDropdownToShow] = useState<string | undefined>()
+  const [loading, setLoading] = useState(true)
   const [filteredApps, setFilteredApps] = useState<NotifyClientTypes.NotifySubscription[]>([])
   const { activeSubscriptions } = useContext(W3iContext)
   const nav = useNavigate()
 
-  const filterApps = useCallback(
-    debounce((searchQuery: string) => {
-      if (!searchQuery) {
-        setFilteredApps(activeSubscriptions)
+  const fetchApps = async (searchQuery: string) => {
+    const newFilteredApps = [] as NotifyClientTypes.NotifySubscription[]
 
-        return
+    from(activeSubscriptions).subscribe({
+      next: app => {
+        if (!loading) {
+          setLoading(false)
+        }
+        const isAppNameMatch = app.metadata.name.toLowerCase().includes(searchQuery.toLowerCase())
+        if (isAppNameMatch) {
+          newFilteredApps.push(app)
+        }
+      },
+      complete: () => {
+        setFilteredApps(newFilteredApps)
       }
+    })
+  }
 
-      const newFilteredApps = [] as NotifyClientTypes.NotifySubscription[]
+  const searchApps = debounce((searchQuery: string) => {
```

Comment:
@devceline Separated search and fetch functions since we don't need debouncing for the initial fetch, and updated `useEffect` dependencies.

Reply:
I think something here caused a regression in UI/UX

On `app.web3inbox.com`:
- Go to `/notifications`
- Go to `/settings`
- Go back to `/notifications`
- Subs will be there instantly

On this preview:
- Go to `/notifications`
- Go to `/settings`
- Go back to `/notifications`
- There is loading

Reply:
Fixed